core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,323 @@
+                $scope.solutionIsSpecified = (
+                  JSON.stringify(
+                    stateSolutionService.savedMemento) ? true : false);
+              } else {
+                alertsService.addInfoMessage('That solution was incorrect!');
+                $scope.openAddSolutionModal();
+              }
+            });
+        }
+        catch (e) {
+          alertsService.addInfoMessage('That solution was invalid!');
+          $scope.openAddSolutionModal();
+        }
+        stateSolutionService.displayed = result.solution;
+      });
+    };
+
+    $scope.deleteSolution = function(evt) {
+      // Prevent clicking on the delete button from also toggling the display
+      // state of the answer group.
+      evt.stopPropagation();
+
+      alertsService.clearWarnings();
+      $modal.open({
+        templateUrl: 'modals/deleteSolution',
+        backdrop: true,
+        controller: [
+          '$scope', '$modalInstance', function($scope, $modalInstance) {
+            $scope.reallyDelete = function() {
+              $modalInstance.close();
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function() {
+        stateSolutionService.displayed = null;
+        stateSolutionService.saveDisplayedValue();
+        $scope.solutionIsSpecified = false;
+      });
+    };
+
+    $scope.onComponentSave = function() {
+      alertsService.clearWarnings();
+      var explorationId =
+        explorationContextService.getExplorationId();
+      var currentStateName =
+        editorContextService.getActiveStateName();
+      var state = explorationStatesService.getState(currentStateName);
+      var interactionId = stateInteractionIdService.savedMemento;
+      var rulesServiceName =
+        angularNameService.getNameOfInteractionRulesService(
+          interactionId);
+      // Inject RulesService dynamically.
+      var rulesService = $injector.get(rulesServiceName);
+      var answer = stateSolutionService.savedMemento.correctAnswer;
+
+      try {
+        AnswerClassificationService.getMatchingClassificationResult(
+          explorationId, state, answer, true, rulesService).then(
+            function (result) {
+              if (editorContextService.getActiveStateName() !== (
+              result.outcome.dest)) {
+                stateSolutionService.savedMemento.correctAnswer = answer;
+                stateSolutionService.displayed = (
+                  stateSolutionService.savedMemento);
+                stateSolutionService.saveDisplayedValue();
+                $scope.solutionIsSpecified = (
+                  JSON.stringify(
+                    stateSolutionService.displayed) ? true : false);
+              } else {
+                alertsService.addInfoMessage('That solution was incorrect!');
```

Comment:
Do we want to convey why we think it's incorrect? Maybe the real thing you want to convey is that it's inconsistent with the answer groups -- or won't lead to the learner moving on to a new state, since maybe the correct thing for the creator to do is to fix the answer groups rather than the solution.

Btw, do we show a warning next to incorrect solutions / does this affect exploration validation and the warnings in the state graph?